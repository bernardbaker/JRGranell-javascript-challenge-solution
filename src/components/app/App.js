import React, { Component } from 'react';
import './App.scss';

import Wrapper from '../map/ResourceMap';

class App extends Component {
  render() {
    return (
      <div key="app" className="app-container">
        <Wrapper key="resourceMap" />
      </div>
    );
  }
}

export default App;
