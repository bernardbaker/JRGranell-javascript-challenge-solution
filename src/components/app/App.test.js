import React from 'react';
import ReactDOM from 'react-dom';
import { shallow } from 'enzyme';

import App from './App';
import GoogleApiWrapper from '../map/ResourceMap'

describe('App', () => {

  let wrapper;
  beforeEach(() => wrapper = shallow(<App />));

  it('should render correctly', () => expect(wrapper).toMatchSnapshot());

  it('should render a <div />', () => {
    expect(wrapper.find('div').length).toEqual(1);
  });

  it('should render the resource map component', () => {
    expect(wrapper.containsMatchingElement(<GoogleApiWrapper />)).toEqual(true)
  })

});
