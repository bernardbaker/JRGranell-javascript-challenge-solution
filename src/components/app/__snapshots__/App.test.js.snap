// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`App should render correctly 1`] = `
ShallowWrapper {
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <App />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "checkPropTypes": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateError": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": "app",
    "nodeType": "host",
    "props": Object {
      "children": <Connect(Wrapper) />,
      "className": "app-container",
    },
    "ref": null,
    "rendered": Object {
      "instance": null,
      "key": "resourceMap",
      "nodeType": "class",
      "props": Object {},
      "ref": null,
      "rendered": null,
      "type": [Function],
    },
    "type": "div",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": "app",
      "nodeType": "host",
      "props": Object {
        "children": <Connect(Wrapper) />,
        "className": "app-container",
      },
      "ref": null,
      "rendered": Object {
        "instance": null,
        "key": "resourceMap",
        "nodeType": "class",
        "props": Object {},
        "ref": null,
        "rendered": null,
        "type": [Function],
      },
      "type": "div",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
        "legacyContextMode": "parent",
        "lifecycles": Object {
          "componentDidUpdate": Object {
            "onSetState": true,
          },
          "getChildContext": Object {
            "calledByRenderer": false,
          },
          "getDerivedStateFromProps": Object {
            "hasShouldComponentUpdateBug": false,
          },
          "getSnapshotBeforeUpdate": true,
          "setState": Object {
            "skipsComponentDidUpdateOnNullish": true,
          },
        },
      },
    },
  },
  Symbol(enzyme.__childContext__): null,
}
`;
